### PR TITLE
Dockerizing

### PR DESCRIPTION
**Pull request by hellen-22**
I am trying to dockerize but i'm getting errors. I have commented out the database part due to the errors. Some guidance needed.

### CODE REVIEW

**Comment by JimmyTron** [comment]
What nature is the error you are having

**Comment by hellen-22** [comment]
The database is not connecting
I'm getting this error - django.db.utils.OperationalError: could not translate host name "db" to address: Unknown host

**Review by JimmyTron — CHANGES_REQUESTED** [review]
Lets first get it right before we can approve it for merging

File: CMS/settings.py [hunk 1]
```diff
@@ -82,6 +83,20 @@
     }
 }
 
+"""
+
+DATABASES = {
```
[the remark this answers is not in the document]
Comment (sangkips) [reply CMS/settings.py:88]: Can you paste your Dockerfile and docker-compose here?

[the remark this answers is not in the document]
Comment (hellen-22) [reply CMS/settings.py:88]: Dockerfile: 
FROM python:alpine

ENV PIP_DISABLE_PIP_VERSION_CHECK=1
ENV PYTHONDONTWRITEBYTECODE=1
ENV PYTHONUNBUFFERED=1

WORKDIR /code
COPY requirements.txt .

RUN pip install -r requirements.txt
COPY . .

Docker compose.
version: "3.9"

services:
  db:
    image: postgres:alpine
    volumes: 
      - postgres_data:/var/lib/postgresql/data/
    environment: 
      - POSTGRES_DB=postgres
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=postgres

  web:
    build: .
    command: python manage.py runserver 0.0.0.0:8000
    volumes:
      - .:/code
    ports:
      - 8000:8000
    environment:
      - POSTGRES_NAME=postgres
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=postgres
    depends_on:
      - db

volumes:
  postgres_data:

[the remark this answers is not in the document]
Comment (sangkips) [reply CMS/settings.py:88]: Also I have a different opinion on using alpine images when it comes to building Python images. Linux distributions use the GNU version (glibc) of the standard C library that is required by pretty much every C program, including Python. But Alpine Linux uses musl, those binary wheels are compiled against glibc, and therefore Alpine disabled Linux wheel support.

[the remark this answers is not in the document]
Comment (hellen-22) [reply CMS/settings.py:88]: I changed to just a python image but still getting errors.. Would you try dockerizing and see if you'll be successful then push we see.

[the remark this answers is not in the document]
Comment (sangkips) [reply CMS/settings.py:88]: Can you be in a pposition to build docker image alone from the docker file? Like running, `docker build . -t sangkips/django-blog:latest`

[the remark this answers is not in the document]
Comment (sangkips) [reply CMS/settings.py:88]: ## Try this Dockerfile
``` FROM python:3.11-slim-bullseye

ENV PYTHONUNBUFFERED=1

WORKDIR /usr/src/app

RUN apt update && apt install -y g++ libpq-dev gcc musl-dev

COPY requirements.txt .

RUN python3 -m pip install -r requirements.txt

COPY . . ```

[the remark this answers is not in the document]
Comment (sangkips) [reply CMS/settings.py:88]: Before you even create a `docker-compose.yml` file just check if that dockerfile can build successfully

[the remark this answers is not in the document]
Comment (sangkips) [reply CMS/settings.py:88]: Where you get the images go to `dockerhub.com` and search the image you want to use

[the remark this answers is not in the document]
Comment (Fredrick-Waihenya) [reply CMS/settings.py:88]: try changing the image and add this command to the ``docker-compose.yml ``:
```web:
    build: .
    command: bash -c "python manage.py migrate && 
                      python manage.py runserver 0.0.0.0:8000"
```

[the remark this answers is not in the document]
Comment (hellen-22) [reply CMS/settings.py:89]: This worked

**Comment by nyandika** [comment]
Is there a reason to be dockerizing at this point?
I can understand the use case for it, but for people getting started into the project with not much experience in the area I dont particularly see if the benefits outweigh what could possibly be a steeper learning curve

**Pull request closed without merging**